lib/vector-core/src/event/vrl_target.rs

```diff
@@ -71,12 +79,17 @@ impl vrl_lib::Target for VrlTarget {
                 .insert(path.clone(), value)
                 .map(|_| ())
                 .map_err(|err| err.to_string()),
-            VrlTarget::Metric(ref mut metric) => {
+            VrlTarget::Metric {
+                ref mut metric,
+                value: metric_value,
+            } => {
                 if path.is_root() {
                     return Err(MetricPathError::SetPathError.to_string());
                 }
 
                 if let Some(paths) = path.to_alternative_components(MAX_METRIC_PATH_DEPTH).get(0) {
+                    let new = value.clone();
```

Comment:
Do we need this `new` or could `value.clone()` be moved down into line 138? It's not totally clear to me if `new` was supposed to have been in use prior to that line.

Reply:
Good catch. While moving it down doesn't work — because all the `try_*` calls take `value` by-value — I can at least move the clones as close as possible to those calls.

It does make me wonder whether we should change the `try_*` methods to take self by reference, and then clone internally _if_ the conversion succeeds. This would forgo cloning if the error-case is hit. I'll think some more on this for follow-up PRs.

Resolved in [6632cca](https://github.com/vectordotdev/vector/pull/12546/commits/6632cca500b8c54ba360d40b8f8aa64acd4b352f).

Reply:
> It does make me wonder whether we should change the try_* methods to take self by reference, and then clone internally if the conversion succeeds. This would forgo cloning if the error-case is hit. I'll think some more on this for follow-up PRs.

I think this would be a really compelling change.